.github/workflows/dotnet.yml

```diff
@@ -0,0 +1,39 @@
+name: Chorus .NET6
+
+on:
+  [push, pull_request, workflow_dispatch]
+
+jobs:
+  build_test_deploy:
+    env:
+      NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    permissions: write-all
+    
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0 # fetch full history for GitVersion
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        source-url: https://nuget.pkg.github.com/sillsdev/index.json
+    - name: Fix Python on Linux (for mercurial)
+      if: matrix.os == 'ubuntu-latest'
+      run: sudo apt install python-is-python2
+    - name: Restore
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --no-restore -c Release
+    - name: Test
+      run: dotnet test
+    - name: Pack
+      if: github.ref == 'refs/heads/master'
+      run: dotnet pack --no-restore --no-build -c Release
+    - name: Deploy
+      if: github.ref == 'refs/heads/master'
+      run: dotnet nuget push output\*.nupkg -s https://api.nuget.org/v3/index.json -k ${{secrets.SILLSDEV_PUBLISH_NUGET_ORG}} --skip-duplicate
```

Comment:
We should also add a step that archives the artifacts so they can be used for testing. See the ["Upload Artifacts" step e.g. in icu-dotnet](https://github.com/sillsdev/icu-dotnet/blob/master/.github/workflows/CI-CD.yml).

If you want to get fancy you could add some additional steps as found in icu-dotnet workflow: archiving test results and displaying them in a separate workflow (the event file is needed for that), and doing the publish to nuget as separate job.